hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
I'm not aware of any limits from CDI. My guess is that it would just wait. I had something similar for the Cassandra `Session` on Accounts, and I never had issues. 

But actually, you might have just found the original issue: the usage of `@Initialized` is new, isn't it? Is it possible that CDI's `@Initialized` event makes this method trigger far before than the previous implementation? How was it previously done? Was it with `@Singleton @Startup` EJB?